src/mrpro/data/SpatialDimension.py

```diff
@@ -24,80 +46,401 @@ class XYZ(Protocol[T]):
+        return (self.x < other.x) & (self.y < other.y) & (self.z < other.z)
+
+    @overload
+    def __le__(self: SpatialDimension[T_co_vector], other: SpatialDimension[T_co_vector]) -> T_co_vector: ...
+    @overload
+    def __le__(self: SpatialDimension[T_co_scalar], other: SpatialDimension[T_co_scalar]) -> bool: ...
+    def __le__(
+        self: SpatialDimension[T_co_scalar] | SpatialDimension[T_co_vector],
+        other: SpatialDimension[T_co_scalar] | SpatialDimension[T_co_vector],
+    ) -> bool | T_co_vector:
+        """Check if self is less than or equal to other."""
+        if not isinstance(other, SpatialDimension):
+            return NotImplemented
+        return (self.x <= other.x) & (self.y <= other.y) & (self.z <= other.z)
+
+    @overload
+    def __gt__(self: SpatialDimension[T_co_vector], other: SpatialDimension[T_co_vector]) -> T_co_vector: ...
+    @overload
+    def __gt__(self: SpatialDimension[T_co_scalar], other: SpatialDimension[T_co_scalar]) -> bool: ...
+    def __gt__(
+        self: SpatialDimension[T_co_scalar] | SpatialDimension[T_co_vector],
+        other: SpatialDimension[T_co_scalar] | SpatialDimension[T_co_vector],
+    ) -> bool | T_co_vector:
+        """Check if self is greater than other."""
+        if not isinstance(other, SpatialDimension):
+            return NotImplemented
+        return (self.x > other.x) & (self.y > other.y) & (self.z > other.z)
+
+    @overload
+    def __ge__(self: SpatialDimension[T_co_vector], other: SpatialDimension[T_co_vector]) -> T_co_vector: ...
+    @overload
+    def __ge__(self: SpatialDimension[T_co_scalar], other: SpatialDimension[T_co_scalar]) -> bool: ...
+    def __ge__(
+        self: SpatialDimension[T_co_scalar] | SpatialDimension[T_co_vector],
+        other: SpatialDimension[T_co_scalar] | SpatialDimension[T_co_vector],
+    ) -> bool | T_co_vector:
+        """Check if self is greater than or equal to other."""
+        if not isinstance(other, SpatialDimension):
+            return NotImplemented
+        return (self.x >= other.x) & (self.y >= other.y) & (self.z >= other.z)
+
+    def __post_init__(self):
+        """Ensure that the data is of matching shape."""
+        if not all(isinstance(val, (int | float)) for val in self.zyx):
+            try:
+                zyx = [_as_vectortype(v) for v in self.zyx]
+                self.z, self.y, self.x = torch.broadcast_tensors(*zyx)
+            except RuntimeError:
+                raise ValueError('The shapes of the tensors do not match') from None
+
+    @property
+    def shape(self) -> tuple[int, ...]:
+        """Get the shape of the x, y, and z.
+
+        Returns
+        -------
+            Empty tuple if x, y, and z are scalar types, otherwise shape
+
+        Raises
+        ------
+            ValueError if the shapes are not equal
+        """
+        if isinstance(self.x, ScalarTypes) and isinstance(self.y, ScalarTypes) and isinstance(self.z, ScalarTypes):
+            return ()
```

Comment:
would I not expect an error if I call shape on a scalar?

Reply:
Calling shape on scalar numpy arrays or tensors returns also an empty tuple.